library/proc_macro/src/lib.rs

```diff
@@ -1200,3 +1200,17 @@ pub mod tracked_env {
         value
     }
 }
+
+/// Tracked access to additional files.
+#[unstable(feature = "track_path", issue = "73921")]
+pub mod tracked_path {
```

Comment:
We should ideally reduce the stutter here `tracked_path::path` (in contrast to `tracked_env::var` above). I understand the above tries to mimick the `std::env::var` but it's not so clear cut with the FS operations (`File::open`, `OpenOptions::open`, `fs::read_to_string` etc.).

Since `mod tracked_env` is unstable, we might group them under a common `mod track` but that seems somewhat build system centric and probably wouldn't be ideal if we decide to add a `lex_file(&Path) -> Result<TokenStream>` or similar.

AIUI there's benefit in providing a familiar API to `std::fs` to promote migrating any FS operations in proc macro to this new proc-macro API - this facilitates build input analysis but could also potentially allow for more sandboxed I/O operation if we, let's say, prune or somehow limit `std::fs` in proc macros.

Reply:
We could move to a submodule structure `tracked::{fs::path, env::var}` which would allow extension as needed a at least vaguely resemble `std::` module structures.

Reply:
I went ahead and restructured it that way, it seemed the best path fwd.

Reply:
I would prefer to avoid changing other APIs like `tracked_env` in this PR.
The module structure can be rearranged later during stabilization.

Reply:
I agree; if we want to restructure this, I think we should do this in a separate, follow-up PR.